sdk/python/setup.py

```diff
@@ -60,6 +65,54 @@ def find_version(*file_path_parts):
   raise RuntimeError('Unable to find version string.')
 
 
+KFPSDK_DIR = os.path.realpath(os.path.dirname(__file__))
+
+# Find the Protocol Compiler. (Taken from protobuf/python/setup.py)
+if "PROTOC" in os.environ and os.path.exists(os.environ["PROTOC"]):
+    PROTOC = os.environ["PROTOC"]
+else:
+    PROTOC = find_executable("protoc")
+
+def GenerateProto(source):
+    """Generate a _pb2.py from a .proto file.
+
+    Invokes the Protocol Compiler to generate a _pb2.py from the given
+    .proto file.  Does nothing if the output already exists and is newer than
+    the input.
+
+    Args:
+      source: The source proto file that needs to be compiled.
+    """
+
+    output = source.replace(".proto", "_pb2.py")
+
+    if not os.path.exists(output) or (
+            os.path.exists(source) and
+            os.path.getmtime(source) > os.path.getmtime(output)):
+        print("Generating %s..." % output)
+
+        if not os.path.exists(source):
+            sys.stderr.write("Can't find required file: %s\n" % source)
+            sys.exit(-1)
+
+        if PROTOC is None:
+            sys.stderr.write(
+                "protoc is not found.  Please compile it "
+                "or install the binary package.\n"
+            )
+            sys.exit(-1)
+
+        protoc_command = [PROTOC, "-I%s" % KFPSDK_DIR, "--python_out=.", source]
+        if subprocess.call(protoc_command) != 0:
+            sys.exit(-1)
+
+
+# Generate the protobuf files that we depend on.
```

Comment:
Let's check in a generated python proto file, so that the users do not need to have protoc installed.
Let's also add a shell script alongside the proto which will re-generate the python proto file if someone changes it.

Reply:
I do some concerns re: checking in generated code in source control. The reasons are most along the line of [1]

I don't think protoc itself is a very heavy dependency for users if they want to contribute to our codebase, and by encoding that into `setup.py` the local dev experience is almost the same as before.

[1] https://softwareengineering.stackexchange.com/questions/192113/do-i-check-generated-code-in-to-source-control-or-not